app/src/editor/canvas/components/Ports/Port/index.jsx

```diff
@@ -0,0 +1,174 @@
+// @flow
+
+import React, { useCallback, useState, useEffect } from 'react'
+import cx from 'classnames'
+import startCase from 'lodash/startCase'
+import EditableText from '$shared/components/EditableText'
+import { DragDropContext } from '../../DragDropContext'
+import Option from '../Option'
+import Plug from '../Plug'
+import Menu from '../Menu'
+import Value from '../Value'
+import Cell from './Cell'
+import styles from './port.pcss'
+
+type Props = {
+    api: any,
+    canvas: any,
+    onPort: any,
+    onValueChange: (any, any) => void,
+    port: any,
+    setOptions: any,
+}
+
+const Port = ({
+    api,
+    canvas,
+    port,
+    onPort,
+    onValueChange: onValueChangeProp,
+    setOptions,
+}: Props) => {
+    const isRunning = canvas.state === 'RUNNING'
+    const isInput = !!port.acceptedTypes
+    const isParam = 'defaultValue' in port
+    const hasInputField = isParam || port.canHaveInitialValue
+    const [contextMenuTarget, setContextMenuTarget] = useState(null)
+    const [editingName, setEditingName] = useState(false)
+
+    const onContextMenu = useCallback((e: SyntheticMouseEvent<EventTarget>) => {
+        e.preventDefault()
+        // $FlowFixMe wtf?
+        setContextMenuTarget(e.currentTarget)
+    }, [setContextMenuTarget])
+
+    const plug = (
+        <Plug
+            api={api}
+            canvas={canvas}
+            onContextMenu={onContextMenu}
+            port={port}
+            register={onPort}
+        />
+    )
+
+    const dismiss = useCallback(() => {
+        setContextMenuTarget(null)
+    }, [])
+
+    const onDocumentClick = useCallback((e: SyntheticMouseEvent<EventTarget>) => {
+        if (contextMenuTarget && e.target instanceof HTMLElement) {
```

Comment:
Is it possible that `e.target` is an `SVGElement` rather than an `HTMLElement`? `classList` comes from `Element` which is the base class for both.

Reply:
Also, is it possible that `e.target` comes from a different realm i.e. different `HTMLElement` instance? e.g. an iframe.

Reply:
We may have a module that has an `iframe`. Added 8b4611f and changed `HTMLElement` to `Element`.